src/text/Word2Vec.ts

```diff
@@ -0,0 +1,126 @@
+import * as tf from '@tensorflow/tfjs';
+import { createData, createUniqueWord, objectLength, skipGram, UniqueWord } from './word2vec-utils';
+
+export default class Word2Vec {
+  public weight: number[][] | undefined = undefined;
+  public uniqueWords: UniqueWord;
+  private data: number[][];
+  private label: number[];
+  private nwords: number;
+  private size: number;
+
+  /**
+   *
+   * @param text Array of strings
+   * @param window window size for skip-gram
+   * @param stopwords
+   * @param size embedding dimension
+   */
+  constructor(text: string[], window: number, stopwords: string[] | undefined, size = 50) {
+    const [ wordList, allText ] = skipGram(window, text, stopwords);
+    this.uniqueWords = createUniqueWord(allText as string[]);
+    this.nwords = objectLength(this.uniqueWords);
+    const [ data, label ] = createData(wordList as string[][], this.uniqueWords, this.nwords);
+    this.data = data as number[][];
+    this.label = label as number[];
+    this.size = size;
+  }
+
+  /**
+   * Train word2vec model
+   * @param epochs
+   * @param verbose to print out the training logs or not
+   * @return Promise
+   */
+  public async train(epochs = 100, verbose = 0): Promise<void> {
+    const model = tf.sequential();
+    model.add(tf.layers.dense({ units: this.size, inputShape: [ this.nwords ] }));
+    model.add(tf.layers.dense({ units: this.nwords, activation: 'softmax' }));
+    model.compile({ optimizer: 'sgd', loss: 'sparseCategoricalCrossentropy', metrics: [ 'accuracy' ] });
```

Comment:
The model might be moved in the constructor? Consider someone calls train twice :)

Reply:
That was my initial method, but I don't get how to use `async` with class `constructor`.

Reply:
We might need to create the model and add layers at the constructor, which doesn't require await anything :)

Reply:
:thinking: the main reason why I even need the aync/wait is that after training, I will need to update the `word2vec weight`. and with this initial approach, even when someone calls `train` twice, the `weight` will never add up, it will always look like it's being called once.

What do you think? Maybe did not get what you are trying to say.

Reply:
Okay, it makes sense, resolved.